pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -509,8 +512,8 @@ func (c *Reconciler) reconcile(ctx context.Context, pr *v1beta1.PipelineRun, get
 	// Ensure that the array reference is not out of bound
 	if err := resources.ValidateParamArrayIndex(ctx, pipelineSpec, pr); err != nil {
 		// This Run has failed, so we need to mark it as failed and stop reconciling it
-		pr.Status.MarkFailed(ReasonObjectParameterMissKeys,
-			"PipelineRun %s/%s parameters is missing object keys required by Pipeline %s/%s's parameters: %s",
+		pr.Status.MarkFailed(ReasonParamArrayIndexingInvalid,
+			"PipelineRun %s/%s array indexing params fail validation by Pipeline %s/%s's parameters: %s",
```

Comment:
The error message looks confusing, could you please rephrase it a little bit? Is this function `ValidateParamArrayIndex` only for validating the out-of-boundary error? if so, can we make it clearer?

Thank you!

Reply:
Proposed codes suggestions has been merged, so I guess we can resolve this comment thread :) Thank you